charts/travis-build/travis-build.yaml

```diff
@@ -0,0 +1,87 @@
+apiVersion: v1
```

Comment:
I'm not sure this file is supposed to be committed. :)

Reply:
why wouldn't it be? legitimate question, a bit new to this stuff, but it looks like a metadata file that would be useful to have? 🤷‍♀️

Reply:
This looks like a dump of a specific pod that was already loaded into Kubernetes. It looks like the result of running `kubectl get pod travis-build-f499bdbb6-67td6 -o yaml` and saving it to a file for inspection. It's got a number of fields that are not generally assigned by users, but rather by the PodController that is managing the pod (things like timestamps, statuses, docker container IDs).

We also generally wouldn't deal in pods directly for the purposes of configuring applications to run in the cluster. We would create Deployments, which manage ReplicaSets, which manage the actual Pods. This pod looks like it was originally created by a Deployment, as it has two levels of random identifiers added to its name (one to distinguish the replicasets, one to distinguish the pods).

Reply:
👍 sgtm

Reply:
@mjm Indeed, this is exactly how the file was generated. Is the recommended way to do it with a `kind: Deployment` file instead of `kind: Pod`?

Reply:
I'm a little confused by this question. I don't think this file needs to be committed. The Helm chart will generate the right Kubernetes resources from the `templates/` directory, so this file wouldn't be used when deploying travis-build.

In terms of writing templates, though, we do want to use `kind: Deployment` as a basic rule for getting our pods into Kubernetes. Even if it's only going to be one replica, it offers a bunch of nice features around keeping the pods alive and managing rollouts of new versions.